sdk/azidentity/errors.go

```diff
@@ -101,24 +101,31 @@ func (*AuthenticationFailedError) NonRetriable() {
 
 var _ errorinfo.NonRetriable = (*AuthenticationFailedError)(nil)
 
-// credentialUnavailableError indicates a credential can't attempt
-// authentication because it lacks required data or state.
+// credentialUnavailableError indicates a credential can't attempt authentication because it lacks required
+// data or state
 type credentialUnavailableError struct {
-	credType string
-	message  string
+	message string
 }
 
+// newCredentialUnavailableError is an internal helper that ensures consistent error message formatting
 func newCredentialUnavailableError(credType, message string) error {
-	return &credentialUnavailableError{credType: credType, message: message}
+	msg := fmt.Sprintf("%s: %s", credType, message)
+	return &credentialUnavailableError{msg}
 }
 
-func (e *credentialUnavailableError) Error() string {
-	return e.credType + ": " + e.message
+// NewCredentialUnavailableError constructs an error indicating a credential can't attempt authentication
+// because it lacks required data or state. When [ChainedTokenCredential] receives this error it will try
+// its next credential, if any.
+func NewCredentialUnavailableError(message string) error {
```

Comment:
I know I had the parameter a `string` in my initial comment, but now I'm wondering if `error` might be better?  Of course, in the case of an error that prompts code to call this ctor, the caller can always pass `err.Error()` as the parameter.  Mostly I'm thinking about this in terms of versioning.  My gut tells me `error` but I could be convinced otherwise.  What do you think?

Reply:
What sort of versioning jam do you have in mind? I'm struggling to come up with a case in which we need more from an error than its string.

Reply:
I don't have anything concrete at present and am probably just overthinking it.  Really, all we need to convey to the token chain is "log this then move along" and this does exactly that.